user_interaction.py

```diff
@@ -0,0 +1,137 @@
+"""Functions for interaction with users, such as get_user_input"""
+import logging
+from typing import Any, List, Optional
+
+from colorama import Back, Fore
+
+from config import IS_DEBUG
+from music_api import ReleaseRecording
+
+logging.basicConfig()
+logger = logging.getLogger("XP3")
+logger.setLevel(logging.DEBUG if IS_DEBUG else logging.INFO)
+
+
+def get_user_input(prompt: str, default: Any) -> str:
+    """Handles user input for functions that require interactivity
+
+    Args:
+        prompt (str): The message who'll be shown to the user
+        default (Any): The default value to be suggested to the user
+
+    Returns:
+        str: The chosen user input
+    """
+    if prompt.endswith(":"):
+        prompt = prompt[:-1]
+    prompt = prompt.strip()
+    if default or default == 0:
+        if str(default).lower() == "true":
+            prompt = f"{prompt} [Y/n]"
+        elif str(default).lower() == "false":
+            prompt = f"{prompt} [y/N]"
+        else:
+            prompt = f"{prompt} [{default}]"
+    user_input = input(prompt + ": ")
+
+    if not user_input:
+        return default
+    chosen_option = user_input
+
+    # Convert to booleans
+    if chosen_option.lower() in ("y", "yes"):
+        chosen_option = True
+    if chosen_option.lower() in ("n", "no"):
+        chosen_option = False
+
+    return chosen_option
+
+
+# TODO - consider changing this with `pick`
+def print_suggestions(
+    recordings: List[ReleaseRecording],
+    artist: str,
+    title: str,
+    suggested_recording_index: int,
+):
```

Comment:
maybe genricify this? i could see a future where you need to print suggestions of a List that is not ReleaseRecording.

Reply:
on the other hand i see that there is some logic that is dependent on the type itself... maybe dependency injection (i.e. the function recieves a type and functions to print that specific type so the function user is actually responsible to how it is printed and all this function does is allowing the client to choose an option

Reply:
maybe thats overcomplicating it though, your call

Reply:
This might get replaced with the pick library later on.
I don't think it's worth the effort at the moment